rust/parquet/src/util/cursor.rs

```diff
@@ -129,6 +132,56 @@ impl Seek for SliceableCursor {
     }
 }
 
+/// Use this type to write Parquet to memory rather than a file.
+#[derive(Debug, Default, Clone)]
+pub struct WriteableCursor {
```

Comment:
nit: it's not easy to see this writes to memory - maybe we can make it more explicit in naming like `InMemoryWriteableCursor` or similar?